nebula_ros/src/continental/continental_srr520_ros_wrapper.cpp

```diff
@@ -0,0 +1,288 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <nebula_ros/continental/continental_srr520_ros_wrapper.hpp>
+
+namespace nebula
+{
+namespace ros
+{
+ContinentalSRR520RosWrapper::ContinentalSRR520RosWrapper(const rclcpp::NodeOptions & options)
+: rclcpp::Node(
+    "continental_srr520_ros_wrapper", rclcpp::NodeOptions(options).use_intra_process_comms(true)),
+  wrapper_status_(Status::NOT_INITIALIZED),
+  packet_queue_(3000),
+  hw_interface_wrapper_(),
+  decoder_wrapper_()
+{
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+
+  wrapper_status_ = DeclareAndGetSensorConfigParams();
+
+  if (wrapper_status_ != Status::OK) {
+    throw std::runtime_error(
+      (std::stringstream{} << "Sensor configuration invalid: " << wrapper_status_).str());
+  }
+
+  RCLCPP_INFO_STREAM(get_logger(), "SensorConfig:" << *config_ptr_);
+
+  launch_hw_ = declare_parameter<bool>("launch_hw", param_read_only());
+
+  if (launch_hw_) {
+    hw_interface_wrapper_.emplace(this, config_ptr_);
+    decoder_wrapper_.emplace(this, config_ptr_, hw_interface_wrapper_->HwInterface());
+  } else {
+    decoder_wrapper_.emplace(this, config_ptr_, nullptr);
+  }
+
+  RCLCPP_DEBUG(get_logger(), "Starting stream");
+
+  decoder_thread_ = std::thread([this]() {
+    while (true) {
+      decoder_wrapper_->ProcessPacket(std::move(packet_queue_.pop()));
```

Comment:
`std::move` is not needed here and prevents copy elision
```suggestion
      decoder_wrapper_->ProcessPacket(packet_queue_.pop());
```

Reply:
Addressed in [3dc3bf9](https://github.com/tier4/nebula/pull/124/commits/3dc3bf9bc71e2433bebdc27c6300e8c994672d99)